Generate TypeScript decorator signatures from `extern dec` (#2122)
microsoft/typespec: CONTRIBUTING.md

See also below for having this happen automatically in VS Code
whenever you save.

## Generate TypeScript signature for decorators

> For all packages execpt the compiler this will be done automatically as part of each package `build` script.

Reviewer: ```suggestion
For all packages except the compiler this will be done automatically as part of each package `build` script.
```

> If wanting to regen the signatures without the build you can run for any package that has decorators

Reviewer: ```suggestion
If you want to regenerate decorator signatures without a full build you can run:
```
